### PR TITLE
Fixed setsockopt bug for linux

### PR DESCRIPTION
**Pull request by realxxmy**

### CODE REVIEW

File: mono/io-layer/sockets.c [hunk 1]
```diff
@@ -799,6 +799,10 @@ int _wapi_setsockopt(guint32 fd, int level, int optname,
 	int ret;
 	const void *tmp_val;
 	struct timeval tv;
+
+#if defined (__linux__)
```
Comment (Tak) [inline mono/io-layer/sockets.c]: Should check PLATFORM_LINUX instead

**Comment by joncham** [comment]
LGTM. Are you making the changes that @Tak mentioned?

**Comment by realxxmy** [comment]
PLATFORM_LINUX not work corrected

**Comment by joncham** [comment]
Hello @realxxmy , can you please specify in both this PR and the other one that you are contributing these changes under the MIT/X11 license, specifically:

`Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so.`

**Comment by realxxmy** [comment]
yes, i can
@joncham

**Comment by joncham** [comment]
@realxxmy sorry, can you please specify the above text in the two PRs?

**Comment by realxxmy** [comment]
Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so.

**Pull request merged**